### PR TITLE
Adds Krav Maga and Hallway Brawler fighting styles.

### PR DESCRIPTION
**Pull request by Iamgoofball**
kravmaga

gue4gd8
Krav Maga is a fighting style good for 1 on 1 combat. It's obtainable from the traitor uplink in the form of gloves for 10tc, and Round-Start Security staff know it innately. It's got 3 combos, a Leg Sweep, a Quick Choke, a Head Elbow, and a Neck Chop.

You do a base 10 damage per punch, and you do bonus damage to weakened on the ground opponents. You sacrifice your chances to weaken people on hit for this.

When disarming someone, you grab objects they are holding, instead of it hitting the ground. You sacrifice your knockdown with disarm for this.

You also have a 25% chance to either counter or block hits directed at you in melee.

If someone is using Krav Maga, which you can tell by the hit sounds, keep your distance and pelt them from afar with ranged weapons.

This is probably a bit overpowered and could use some nerfing, so I'll let you guys come up with values.

Hallway Brawler is a silly thing I coded for fun while I was figuring out how to use the martial arts datums. I left it in as an admin spawn/thing you beg the admins to let your trade TCs for, as it's pretty fun to play with and play against. It also has the countering and blocking mechanic, and you can find the info on it's combos using the ability the gloves give you. Sprites for this glove are by Ausops.

Sounds are from freesound.org.
https://www.freesound.org/people/RSilveira_88/sounds/216196/
https://www.freesound.org/people/Taira%20Komori/sounds/214200/
https://www.freesound.org/people/FlameEagle/sounds/131142/

### CODE REVIEW

**Comment by SconesC** [comment]
:-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1: :-1:  not for sec you monster

**Comment by Iamgoofball** [comment]
@SconesC Earn your Krav Maga by taking it from the graves of your fallen enemies

**Comment by PKPenguin321** [comment]
dude this looks hella cool and i can definitely see myself taking the traitor one :+1:

**Comment by Fox-McCloud** [comment]
Oh gosh wrasslin' gloves.

:+1:

**Comment by PKPenguin321** [comment]
Also this might be a good alternative fighting style for Gang rather than plasma fist, paging Dr. @Xhuis

**Comment by Iamgoofball** [comment]
@PKPenguin321 it isn't plasma fist nerd

**Comment by KorPhaeron** [comment]
Security starting with combat training isn't a bad idea.

**Comment by Razharas** [comment]
Id prefer they start with something disabling like wrestlingish things instead of damaging

**Comment by RemieRichards** [comment]
you spelt hadouken wrong in the explanation text for Shakunetsu hadouken.

**Comment by KorPhaeron** [comment]
If they've been stripped of all their weapons they're up against something that requires lethal force more often than not

**Comment by Iamgoofball** [comment]
@Razharas If a security officer is to the point of being unarmed he probably needs that lethal force.

**Comment by Ausops** [comment]
>sec trained in krav maga

yesssss

**Comment by bustygoku** [comment]
I like that security is formidable on their own outside of their base gear.

They should be different from normal crew members in that they're ~trained professionals~ or at least pretend to be that way. Mall ninja martial arts fits perfectly.

**Comment by Iamgoofball** [comment]
@SconesC fix your fukken comment nerd

**Comment by PKPenguin321** [comment]
ohhh shit i didn't realize security starts with it too
that's kind of odd considering it looks like it's mostly used for murdering people (your hands become toolboxes) and doesn't seem all that fitting for sec, but as long as traitors get it too that's cool i guess
i'd like to see how this plays out on a live server

**Comment by Iamgoofball** [comment]
@PKPenguin321 It's actually really good for subduing criminals.

**Comment by PKPenguin321** [comment]
yeah see that's what i'm thinking because of the stun moves, but with that comes the ten/fifteen damage fists so there's kind of a tradeoff there
like i said i'd like to see it on a live server to see how it works when handed to players

**Comment by Iamgoofball** [comment]
@PKPenguin321 Yeah, but if they're to the point of just hammering on the guy with their fists it's probably a "do or die" situation anyways

**Comment by Miauw62** [comment]
put giving it to sec into another PR or something vOv

**Comment by Iamgoofball** [comment]
Pretty much everyone and their mom that I've shown it to so far likes the idea of sec getting it, I'll keep it in this pr unless everyone changes their mind.

**Comment by Fox-McCloud** [comment]
Kinda with @Miauw62 on this one, personally---Sec already has access to equipment no one else does, now you're giving something, tied directly to the job, and unobtainable to non-traitors...to them.

*shrug* just my 2 cents.

File: code/datums/martial/hallwaybrawler.dm [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+	item_state = "fightgloves"
+	var/datum/martial_art/hallway_brawler/style = new
+
+/obj/item/clothing/gloves/hallway_brawler/equipped(mob/user, slot)
+	if(!ishuman(user))
+		return
+	if(slot == slot_gloves)
+		var/mob/living/carbon/human/H = user
+		style.teach(H,1)
+		user.verbs += /mob/living/carbon/human/proc/hallway_brawler_help
```
Comment (Zuhayr) [inline code/datums/martial/hallwaybrawler.dm]: Help verbs could be put into the datum teach proc surely?

**Comment by Dorsisdwarf** [comment]
Security has *no* equipment that anyone with a stunprod+wire coil / emag cannot get. I like the idea of roundstart officers being slightly better than some random shithead greytider, able to beat your ass to the ground even if you loldisarm his baton

**Comment by OhEmGee** [comment]
[heavy breathing]

**Comment by Profakos** [comment]
Could there possibly be a config option for servers, in case I want to make all cops be Hallway Brawlers/boxers/none instead?

**Comment by Nanotransistor** [comment]
Who cares about balance. Might as well add a pulse rifle or two to the armory while we are at it.

**Comment by valzargaming** [comment]
This is an amazing PR, Goof!

**Comment by Iamgoofball** [comment]
@Profakos nah, sorry

File: code/datums/martial/hallwaybrawler.dm (hunk 1 above)
Comment (Iamgoofball) [inline code/datums/martial/hallwaybrawler.dm]: probably a good idea, yeah

**Comment by Supermichael777** [comment]
put it on nukeops gloves and im sold

**Comment by MrPerson** [comment]
I'm going to use this as a soap box to remind everyone that the combos are

REALLY

fucking stupid

**Comment by SconesC** [comment]
They're pretty functional in the case of Sleeping Carp and Plasma Fist, MrPerson

What are you talking about?

**Comment by Rinoka** [comment]
@MrPerson I basically thought the same thing and then remembered there's hotkeys for changing intent on the fly, so

git gud

**Comment by PKPenguin321** [comment]
yeah combos work surprisingly well and i'm actually super excited for this to be in the game, i might actually play sec more

**Comment by whitefox08** [comment]
:+1: 

waiting  for raslen belt

**Comment by Iamgoofball** [comment]
@Supermichael777 no, use your tcs to buy it

**Comment by PKPenguin321** [comment]
actually if security has it because they're trained beforehand then it would make sense for ops to have it too

**Comment by Rinoka** [comment]
That or an op-specific one.

**Comment by Iamgoofball** [comment]
removed from sec, FOR ANOTHER PR

adds a changelog

fixed some bugs

File: code/modules/mob/living/carbon/human/species.dm [hunk 2]
```diff
@@ -914,7 +916,11 @@
 		user.do_attack_animation(H)
 	if((user != H) && H.check_shields(I.force, "the [I.name]", I))
 		return 0
-
+	if(H.martial_art)
```
Comment (Aranclanos) [inline code/modules/mob/living/carbon/human/species.dm]: both checks are bad

Comment (Iamgoofball) [inline code/modules/mob/living/carbon/human/species.dm]: no other way as far as I know, you got an alternative?

Comment (Aranclanos) [inline code/modules/mob/living/carbon/human/species.dm]: exactly what I told you on your super 10000 pr

**Comment by Supermichael777** [comment]
no longer sold but traitor does need direct buffs like this. at least it's not a direct sec nerf

**Comment by duncathan** [comment]
He said he'd give it to sec for another PR.

**Comment by Rinoka** [comment]
Most likely mainly because it's a balance problem in terms of how easily you can metaknow your sec. :')

**Comment by palpatine213** [comment]
This is certainly an interesting item for traitors. Pretty much every traitorling is going to grab this, since mutesting + quick choke = ggnore for the target.
Then again, people are saying that they want the game to be more lethal. So, cautiously optimistic I guess.

**Comment by Miauw62** [comment]
1. ling is stupid
2. being both a traitor and a ling is rare

i've honestly given up trying to balance ling as it stands now, and if they want a ggnore takedown they can still mute+stunprod+armblade.

File: code/modules/mob/living/carbon/human/species.dm (hunk 2 above)
Comment (Iamgoofball) [inline code/modules/mob/living/carbon/human/species.dm]: well, i was looking into it, i couldn't pull it off successfully

**Comment by Cheridan** [comment]
straight-up calling them Hadoukens seems pretty rip-offy to me.

File: code/modules/mob/living/carbon/human/species.dm (hunk 2 above)
Comment (Aranclanos) [inline code/modules/mob/living/carbon/human/species.dm]: just find me on irc and ill slowly explain it to you while stabbing myself

**Comment by bustygoku** [comment]
FORNO FIREBALL

**Comment by Falamazeer** [comment]
Well...
Perhaps you should tie it to implants? Loyalty implants could have a benefit, rather than just meaning revs kill you rather than convert you.

And the traitor equivalent could be tied to a purchasable implant, meaning you could "Disarm" the implant by surgery.

**Comment by Iamgoofball** [comment]
@Falamazeer implant code is atrocious

**Comment by Aranclanos** [comment]
ping me later before i procastinate 5ever

**Comment by Supermichael777** [comment]
@Aranclanos Bump.

**Comment by Jordie0608** [comment]
@Iamgoofball Are you still working on this

**Comment by Cheridan** [comment]
he's not

**Comment by PKPenguin321** [comment]
noo

**Comment by Iamgoofball** [comment]
i'll get this re-opened and fixed up when I get back home

**Comment by PKPenguin321** [comment]
>[i'll get this re-opened and fixed up when I get back home](https://www.youtube.com/watch?v=bq54jb_5WCI)

**Comment by Iamgoofball** [comment]
oh is that what I did to this PR

**Comment by Shadowlight213** [comment]
Wew

**Comment by PKPenguin321** [comment]
i have hijacked the code, PR inbound

**Pull request closed without merging**